package.json

```diff
@@ -7,8 +7,8 @@
   "repository": "github:nextstrain/auspice",
   "homepage": "https://www.npmjs.com/package/auspice",
   "engines": {
-    "node": "^12 || ^14 || ^16",
-    "npm": ">=6.9"
+    "node": "^14 || ^16 || ^18",
```

Comment:
I'm running node v18.9.0 / npm 8.9.2 and thus using the new lockfile format 👇 , so we're going to run into the [same bug we ran into for nextstrain.org](https://github.com/nextstrain/nextstrain.org/pull/587) I think? I'm not sure of a good path to mitigate this, other than having the developers + dependebot use the same node versions.

```
npm WARN old lockfile 
npm WARN old lockfile The package-lock.json file was created with an old version of npm,
npm WARN old lockfile so supplemental metadata must be fetched from the registry.
npm WARN old lockfile 
npm WARN old lockfile This is a one-time fix-up, please be patient...
npm WARN old lockfile 
```

Reply:
> having the developers + dependebot use the same node versions

I think this is the ideal solution here, so we can avoid unnecessary diffs like https://github.com/nextstrain/nextstrain.org/pull/587. Based on https://github.com/dependabot/dependabot-core/issues/2643#issuecomment-775474764, it seems like dependabot is smart enough to not mess with the lockfile version so that shouldn't be an issue. We should also should consider what we use in production (e.g. [nextstrain.org is on Node 14](https://github.com/nextstrain/nextstrain.org/blob/adae78a29e34a38aba9bc91bf33d32ba0070976f/package.json#L11), which is why I choose to use it locally).

But it's also nice to have developers test on newer versions like you've been doing with Node v18. Ideally, errors with newer versions should be caught with CI, but there are still gaps such as https://github.com/nextstrain/auspice/pull/1520#discussion_r977097904. I'd say it's fine for you or other developers to use Node v18 as long as you switch back to Node 14 to make any `package-lock.json` updates.

Reply:
Ideally we'd run the server on node v18 as it's slightly faster (due to V8 upgrade), but I don't know if this would be a measurable difference in our case (if we were to measure such things).

P.S. You're correct that [heroku uses the nodejs version listed in the package.json](https://devcenter.heroku.com/articles/nodejs-support#specifying-a-node-js-version).

Reply:
I created a PR to update the server to v18, which requires merging this PR first: https://github.com/nextstrain/nextstrain.org/pull/595

Reply:
Note that after [the PR splitting](https://github.com/nextstrain/auspice/pull/1520#issuecomment-1267359650), I'm using Node.js 14 for manual testing on this PR since it is now scoped to just webpack version changes.